ppmarq1/covid-19-data-hub
# Covid data hub

Hi! I kindly ask you for a review of this project. I have completed this milestone where we were required to:

- Install API.
- Design: Follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
       -  Colors, Typography: font face, size, and weight.
       -  Layout: composition and space between elements. 
       -  Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

- The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) was done in the README.

- Interactions

     -  Home page

    When the page loads, the web app shows the list of categories that could be filtered out by some parameter, for 
    example by the category name.  
    Along with the category name, you will display some numeric values per category that come from the API.
    When the user clicks (or taps) on a category item, the application navigates to the details page.

     -  Details page

     In the details page, the webapp retrieves data from the API to show detailed data in the category.
    When the user clicks on the "Back" button (<), the user navigates to the home page.

- Testing requirements

    - Create unit tests for pure functions in Redux code.
    - Create integration tests for application using the React Testing Library.
    - Use mock access to the API, so that your tests don't send actual requests.
    - Mock the connection to the Redux Store.

- Technical requirements

    -The project is a single-page application (SPA) built with React and Redux.
   - The data retrieved from the API should be stored in the Redux store.
   - It was filter the data that retrieve from the API using a Filter stateless component.
   - Every page (the main page and the pages for each item)  have a unique route within the SPA.
   - The project was deployed and accessible online.

- Project documentation
   - Record a 5 minutes video on Loom explaining the description of the project.
  
Kind Regards,


Pedro Marques
